core/v2/types.go

```diff
@@ -0,0 +1,96 @@
+package corev2
+
+import (
+	"math"
+
+	"github.com/Layr-Labs/eigenda/encoding"
+)
+
+var (
+	ParametersMap = map[uint8]BlobVersionParameters{
+		0: {CodingRate: 8, ReconstructionThreshold: 0.22, NumChunks: 8192},
+	}
+)
+
+type QuorumID = uint8
+
+type OperatorID = [32]byte
+
+// Assignment contains information about the set of chunks that a specific node will receive
+type Assignment struct {
+	StartIndex uint32
+	NumChunks  uint32
+}
+
+// GetIndices generates the list of ChunkIndices associated with a given assignment
+func (c *Assignment) GetIndices() []uint32 {
+	indices := make([]uint32, c.NumChunks)
+	for ind := range indices {
+		indices[ind] = c.StartIndex + uint32(ind)
+	}
+	return indices
+}
+
+// BlobHeader contains all metadata related to a blob including commitments and parameters for encoding
+type BlobHeader struct {
+	Version uint8
+
+	encoding.BlobCommitments
+
+	// QuorumInfos contains the quorum specific parameters for the blob
+	QuorumNumbers []uint8
+
+	// PaymentHeader contains the payment information for the blob
+	PaymentHeader
+
+	// AuthenticationData is the signature of the blob header by the account ID
+	AuthenticationData []byte `json:"authentication_data"`
+}
+
+func (b *BlobHeader) GetEncodingParams() (encoding.EncodingParams, error) {
+
+	params := ParametersMap[b.Version]
+
+	length, err := GetChunkLength(b.Version, uint32(b.Length))
+	if err != nil {
+		return encoding.EncodingParams{}, err
+	}
+
+	return encoding.EncodingParams{
+		NumChunks:   uint64(params.NumChunks),
+		ChunkLength: uint64(length),
+	}, nil
+
+}
+
+type PaymentHeader struct {
```

Comment:
we should probably just use the one in `core`?

Reply:
We can do some reconciliation once one or the other of the PRs is merged.